src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java

```diff
@@ -279,6 +307,24 @@ public void visitToken(DetailAST ast) {
         }
     }
 
+    /**
+     * Checks if ast is annotation argument and should be checked.
+     * @param ast token to check
+     * @return true if element is skipped, false otherwise
+     */
+    private boolean skipAnnotationArgs(DetailAST ast) {
+        return ignoreAnnotation && isChildOf(ast, TokenTypes.ANNOTATION);
+    }
+
+    /**
+     * Checks if ast is annotation element default value and should be checked.
+     * @param ast token to check
+     * @return true if element is skipped, false otherwise
+     */
+    private boolean skipAnnotationDefaults(DetailAST ast) {
+        return ignoreAnnotationElementDefaults && isChildOf(ast, TokenTypes.LITERAL_DEFAULT);
```

Comment:
If you just check `LITERAL_DEFAULT` and no parent, won't it also ignore violations inside a `switch`'s `default` case?

Reply:
No. In switch `default` literal is not parent of slist where magic number can appear, e.g.

```
~/Downloads $ javac A.java
~/Downloads $ cat A.java
class A {
  {   switch (1) {
            case 1:
            default: int b = 10;
      }
  }
}
~/Downloads $ java -jar checkstyle-8.22-all.jar -t A.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|--LITERAL_CLASS -> class [1:0]
|--IDENT -> A [1:6]
`--OBJBLOCK -> OBJBLOCK [1:8]
    |--LCURLY -> { [1:8]
    |--INSTANCE_INIT -> INSTANCE_INIT [2:2]
    |   `--SLIST -> { [2:2]
    |       |--LITERAL_SWITCH -> switch [2:6]
    |       |   |--LPAREN -> ( [2:13]
    |       |   |--EXPR -> EXPR [2:14]
    |       |   |   `--NUM_INT -> 1 [2:14]
    |       |   |--RPAREN -> ) [2:15]
    |       |   |--LCURLY -> { [2:17]
    |       |   |--CASE_GROUP -> CASE_GROUP [3:12]
    |       |   |   |--LITERAL_CASE -> case [3:12]
    |       |   |   |   |--EXPR -> EXPR [3:17]
    |       |   |   |   |   `--NUM_INT -> 1 [3:17]
    |       |   |   |   `--COLON -> : [3:18]
    |       |   |   |--LITERAL_DEFAULT -> default [4:12] <<<<< default
    |       |   |   |   `--COLON -> : [4:19]
    |       |   |   `--SLIST -> SLIST [4:21]
    |       |   |       |--VARIABLE_DEF -> VARIABLE_DEF [4:21]
    |       |   |       |   |--MODIFIERS -> MODIFIERS [4:21]
    |       |   |       |   |--TYPE -> TYPE [4:21]
    |       |   |       |   |   `--LITERAL_INT -> int [4:21]
    |       |   |       |   |--IDENT -> b [4:25]
    |       |   |       |   `--ASSIGN -> = [4:27]
    |       |   |       |       `--EXPR -> EXPR [4:29]
    |       |   |       |           `--NUM_INT -> 10 [4:29] <<<< number
    |       |   |       `--SEMI -> ; [4:31]
    |       |   `--RCURLY -> } [5:6]
    |       `--RCURLY -> } [6:2]
    `--RCURLY -> } [7:0]
```

Reply:
> In switch default literal is not parent of slist where magic number can appear

Where is this `slist` check in the code?

Reply:
There is no need to check for slist, `isChildOf` is enough.

I updated test cases, please take a look.

Reply:
@strkkk I looked at this in detail and it has nothing to do with `SLIST`.
`LITERAL_DEFAULT` is never the parent token for switch group, so it never gets picked up when scanning in `isChildOf`. `CASE_GROUP` is the parent token.
This is why the case isn't an issue.

Reply:
Please check isChildOf(ast, TokenTypes.LITERAL_DEFAULT, TokenTypes.ANNOTATION);
to be exact, it is not a big a performace issue.

Reply:
@romani sorry, I don't get your point.